app/include/dt-bindings/zmk/animation.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+/**
+ * Maps HSL color settings into a single uint32_t value
+ * that can be cast to zmk_color_hsl.
+ */
+#ifdef CONFIG_BIG_ENDIAN
+#define HSL(h, s, l) ((h << 16) + (s << 8) + l)
+#else
+#define HSL(h, s, l) (h + (s << 16) + (l << 24))
+#endif
+
+/**
+ * Animation blending modes
+ */
+#define BLENDING_MODE_NORMAL 0
+#define BLENDING_MODE_MULTIPLY 1
+#define BLENDING_MODE_LIGHTEN 2
+#define BLENDING_MODE_DARKEN 3
+#define BLENDING_MODE_SCREEN 4
+#define BLENDING_MODE_SUBTRACT 5
+
+/**
+ * Animation control commands
+ */
+#define ANIMATION_CMD_TOGGLE 0
+#define ANIMATION_CMD_NEXT 1
+#define ANIMATION_CMD_PREVIOUS 2
+#define ANIMATION_CMD_SELECT 3
+#define ANIMATION_CMD_BRIGHTEN 4
+#define ANIMATION_CMD_DIM 5
+#define ANIMATION_CMD_NEXT_CONTROL_ZONE 6
+#define ANIMATION_CMD_PREVIOUS_CONTROL_ZONE 7
+
+/**
+ * Generic animation command macro
+ */
+#define ANIMATION_CONTROL_CMD(command, zone, param) ((zone << 24) | (command << 16) | (param))
+
+/**
+ * Animation behavior helpers
+ */
+#define ANIMATION_TOGGLE(zone) ANIMATION_CONTROL_CMD(ANIMATION_CMD_TOGGLE, zone, 0)
```

Comment:
Why do commands like `ANIMATION_TOGGLE` take a `zone`? It does not seem like the animation control implementation currently respects this zone parameter, except for `ANIMATION_SELECT`

Reply:
The `zone` param refers to the `animation_control` instance you wish to target with the behavior.

For example, you could have two different `animation_control` instances, one wrapping your per-key LEDs, and another one for keyboard backlight. Let's say these are the only `animation_control` instances you have with indexes `0` and `1` respectively.

So now, using `&animation ANIMATION_TOGGLE(0)` would trigger `ANIMATION_CMD_TOGGLE` on the first instance, toggling only the lights under your keys. While `&animation ANIMATION_TOGGLE(1)` would to the same for the backlight.

I agree it's slightly confusing in the code, mostly due to how DT bindings work and it's not immediately clear where `zone` comes from.